source/dofs/dof_tools_constraints.cc

```diff
@@ -2819,25 +2819,38 @@ namespace DoFTools
 #endif
           }
 
-        WorkStream::run(coarse_grid.begin_active(),
-                        coarse_grid.end(),
-                        std::bind(&compute_intergrid_weights_3<dim, spacedim>,
-                                  std::placeholders::_1,
-                                  std::placeholders::_2,
-                                  std::placeholders::_3,
-                                  coarse_component,
-                                  std::cref(coarse_grid.get_fe()),
-                                  std::cref(coarse_to_fine_grid_map),
-                                  std::cref(parameter_dofs)),
-                        std::bind(&copy_intergrid_weights_3<dim, spacedim>,
-                                  std::placeholders::_1,
-                                  coarse_component,
-                                  std::cref(coarse_grid.get_fe()),
-                                  std::cref(weight_mapping),
-                                  is_called_in_parallel,
-                                  std::ref(weights)),
-                        scratch,
-                        copy_data);
+        WorkStream::run(
+          coarse_grid.begin_active(),
+          coarse_grid.end(),
+          [coarse_component,
```

Comment:
this looks quite ugly. Would you mind moving the definition of the lambdas before ``run()``?